Gems/PythonAssetBuilder/Editor/Scripts/scene_api/scene_data.py

```diff
@@ -113,30 +113,164 @@ def add_prefab_group(self, name, id, json) -> dict:
         self.manifest['values'].append(prefabGroup)
         return prefabGroup
 
-    def mesh_group_select_node(self, meshGroup, nodeName):
-        meshGroup['nodeSelectionList']['selectedNodes'].append(nodeName)
+    def mesh_group_select_node(self, mesh_group: dict, node_name: str) -> None:
+        mesh_group['nodeSelectionList']['selectedNodes'].append(node_name)
 
-    def mesh_group_unselect_node(self, meshGroup, nodeName):
-        meshGroup['nodeSelectionList']['unselectedNodes'].append(nodeName)
+    def mesh_group_unselect_node(self, mesh_group: dict, node_name: str) -> None:
+        mesh_group['nodeSelectionList']['unselectedNodes'].append(node_name)
 
-    def mesh_group_add_advanced_coordinate_system(self, meshGroup, originNodeName, translation, rotation, scale):
-        originRule =  {}
-        originRule['$type'] = 'CoordinateSystemRule'
-        originRule['useAdvancedData'] = True
-        originRule['originNodeName'] = '' if originNodeName is None else originNodeName
+    def mesh_group_add_advanced_coordinate_system(self, mesh_group: dict, origin_node_name: str, translation: object,
+                                                  rotation: object, scale: float) -> None:
+        origin_rule = {
+            '$type': 'CoordinateSystemRule',
+            'useAdvancedData': True,
+            'originNodeName': '' if origin_node_name is None else origin_node_name
+        }
         if translation is not None:
-            originRule['translation'] = translation
+            origin_rule['translation'] = translation
         if rotation is not None:
-            originRule['rotation'] = rotation
+            origin_rule['rotation'] = rotation
         if scale != 1.0:
-            originRule['scale'] = scale
-        meshGroup['rules']['rules'].append(originRule)
-
-    def mesh_group_add_comment(self, meshGroup, comment):
-        commentRule =  {}
-        commentRule['$type'] = 'CommentRule'
-        commentRule['comment'] = comment
-        meshGroup['rules']['rules'].append(commentRule)
+            origin_rule['scale'] = scale
+        mesh_group['rules']['rules'].append(origin_rule)
+
+    def mesh_group_add_comment(self, mesh_group: dict, comment: str) -> None:
+        commentRule = {
```

Comment:
I wonder if this PR should fill out the docs for each method?

Reply:
Ill follow up with another PR to make those changes, I've already got more PRs ready to follow this which make changes in this file, it'll be easier to avoid merge conflicts that way